test/unit/upload.js

```diff
@@ -36,66 +41,218 @@ describe('Upload Methods', function () {
       }
       mockUser = {}
       mockUser.newInstance = sinon.stub().returns(mockInstance)
-      mockArgs = { _user: mockUser, file: 'mockFile.txt' }
-      sinon.stub(utils, 'getRepositoryAndInstance').resolves([ mockArgs, mockInstance ])
+      sinon.stub(utils, 'getRepositoryAndInstance')
       sinon.stub(fs, 'readFile').yieldsAsync(null, mockFileData)
+      sinon.spy(upload, '_createFile')
+      sinon.spy(upload, '_recusivelyCreateDirectories')
     })
 
     afterEach(function () {
       utils.getRepositoryAndInstance.restore()
       fs.readFile.restore()
+      upload._createFile.restore()
+      upload._recusivelyCreateDirectories.restore()
```

Comment:
now that you have tests for `_recursivelyCreateDirectories` below, I don't have any question that it's doing the correct thing now (`_createFile`, etc). You should just be able to stub this function with a `.resolves()` and minimize your tests below. Shouldn't need more than a few I imagine, including that `_recursivelyCreateDirectories` was called w/ the correct arguments!

(I'm crazy right, more tests, more tests! now, less tests! :smile:)